experimental/material_3_demo/lib/main.dart

```diff
@@ -0,0 +1,202 @@
+    switch (screenIndex) {
+      case 0:
+        return ComponentScreen(showNavBottomBar: showNavBarExample);
+      case 1:
+        return const ColorPalettesScreen();
+      case 2:
+        return const TypographyScreen();
+      case 3:
+        return const ElevationScreen();
+      default:
+        return ComponentScreen(showNavBottomBar: showNavBarExample);
+    }
+  }
+
+  PreferredSizeWidget createAppBar() {
+    return AppBar(
+      title: useMaterial3 ? const Text("Material 3") : const Text("Material 2"),
+      actions: [
+        IconButton(
+          icon: useLightMode
+              ? const Icon(Icons.wb_sunny_outlined)
+              : const Icon(Icons.wb_sunny),
+          onPressed: handleBrightnessChange,
+          tooltip: "Toggle brightness",
+        ),
+        IconButton(
+          icon: useMaterial3
+              ? const Icon(Icons.filter_3)
+              : const Icon(Icons.filter_2),
+          onPressed: handleMaterialVersionChange,
+          tooltip: "Switch to Material ${useMaterial3 ? 2 : 3}",
+        ),
+        PopupMenuButton(
+          icon: const Icon(Icons.more_vert),
+          shape:
+              RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
+          itemBuilder: (context) {
+            return List.generate(colorOptions.length, (index) {
+              return PopupMenuItem(
+                  value: index,
+                  child: Wrap(
+                    children: [
+                      Padding(
+                        padding: const EdgeInsets.only(left: 10),
+                        child: Icon(
+                          index == colorSelected
+                              ? Icons.color_lens
+                              : Icons.color_lens_outlined,
+                          color: colorOptions[index],
+                        ),
+                      ),
+                      Padding(
+                          padding: const EdgeInsets.only(left: 20),
+                          child: Text(colorText[index]))
+                    ],
+                  ));
+            });
+          },
+          onSelected: handleColorSelect,
+        ),
+      ],
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      title: 'Material 3',
+      themeMode: useLightMode ? ThemeMode.light : ThemeMode.dark,
+      theme: themeData,
+      home: LayoutBuilder(builder: (context, constraints) {
+        if (constraints.maxWidth < narrowScreenWidthThreshold) {
+          return Scaffold(
+            appBar: createAppBar(),
+            body: Row(children: <Widget>[
+              createScreenFor(screenIndex, false),
+            ]),
+            bottomNavigationBar: NavigationBars(
+              onSelectItem: handleScreenChanged,
+              selectedIndex: screenIndex,
+              isExampleBar: false,
+            ),
+          );
+        } else {
```

Comment:
The if clause returns so an else clause isn't needed here.

Reply:
I think the `else` would be needed when the screen width is greater than 450 pixels:)

Reply:
I meant that this is logically the same:
```dart
if (constraints.maxWidth < narrowScreenWidthThreshold) {
  return Scaffold(
    appBar: createAppBar(),
    body: Row(
      children: <Widget>[
        createScreenFor(ScreenSelected.values[screenIndex], false),
      ],
    ),
    bottomNavigationBar: NavigationBars(
      onSelectItem: handleScreenChanged,
      selectedIndex: screenIndex,
      isExampleBar: false,
    ),
  );
}

return Scaffold(
  appBar: createAppBar(),
  body: SafeArea(
    bottom: false,
    top: false,
    child: Row(
      children: <Widget>[
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 5),
          child: NavigationRailSection(
            onSelectItem: handleScreenChanged,
            selectedIndex: screenIndex
          ),
        ),
        const VerticalDivider(thickness: 1, width: 1),
        createScreenFor(ScreenSelected.values[screenIndex], true),
      ],
    ),
  ),
);
```

Reply:
Oh I got it. You are right. Will fix!!